src/PageHeader/PageHeader.tsx

```diff
@@ -56,19 +62,103 @@ export type PageHeaderProps = {
   as?: React.ElementType | 'header' | 'div'
 } & SxProp
 
-const Root: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, as = 'div'}) => {
-  const rootStyles = {
-    display: 'flex',
-    flexDirection: 'column',
-    // TODO: We used hard-coded values for the spacing and font size in this component. Update them to use new design tokens when they are ready to use.
-    gap: '0.5rem',
-  }
-  return (
-    <Box as={as} sx={merge<BetterSystemStyleObject>(rootStyles, sx)}>
-      {children}
-    </Box>
-  )
-}
+const Root = React.forwardRef<HTMLDivElement, React.PropsWithChildren<PageHeaderProps>>(
+  ({children, sx = {}, as = 'div'}, forwardedRef) => {
+    const rootStyles = {
+      display: 'grid',
+      // We have max 4 columns.
+      gridTemplateColumns: 'auto auto auto 1fr',
+      gridTemplateAreas: `
+      'context-area context-area context-area context-area'
+      'leading-action title-area trailing-action actions'
+      'description description description description'
+      'navigation navigation navigation navigation'
+    `,
+      //  --custom-height is a custom property (passed by sx) that can be used to override the set height.
+      // We don't want these values to be overriden but still want to allow consumers to override them if needed.
+      '&[data-size-variant="large"]': {
```

Comment:
Non-blocking nitpick. I don't think we need to repeat this style for every size. I think we can safely assume that `--customHeight` will be defined and won't need a fallback.

Reply:
I think I see what you mean. When the custom height is defined, the size won't matter anymore. 

However, if I don't repeat 
```
{
  height: `var(--custom-height, ${LARGE_TITLE_HEIGHT})`,
},
```

for every size, and only define once like 

```
'[data-component="PH_LeadingAction"], [data-component="PH_TrailingAction"],[data-component="PH_Actions"], [data-component="PH_LeadingVisual"], [data-component="PH_TrailingVisual"]':
{
    height: 'var(--custom-height)',
},
```

it will have the lower css specificity against the styles have the `data-size-variant` attribute therefore the custom height won't be applied. Let me know if I misunderstand you or if you have any recommendation to keep the css specificity higher witout repeating for every size.